pages/api/self-check/patch.ts

```diff
@@ -0,0 +1,25 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { SelfCheckGroup } from "../models/selfCheckGroup";
+
+const patchSelfCheckByID = async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const { id } = req.query;
+    const selfCheck = await SelfCheckGroup.findById(id).exec();
+
+    if (!selfCheck)
+        return res
+            .status(404)
+            .send("The self check with the given ID was not found.");
+
+    for (const key in req.body) {
+        if (selfCheck[key] && selfCheck[key] !== req.body[key])
```

Comment:
Should we throw an error if we get a key that's not in the selfcheck model? One thing to look out for if we do this is how it handles optional fields. Also are optional fields correctly handled right now?